src/Components/bookHub/BookHub.js

```diff
@@ -14,20 +17,77 @@ const base = new Airtable({ apiKey: airtableConfig.apiKey })
 
 function CardsDisplay() {
   const [cards, setCards] = useState([]);
+  const [filteredCards, setFilteredCards] = useState([]);
+  const [rangeInput, setRangeInput] = useState({
+    age: { min: ageRangeMetadata[0], max: ageRangeMetadata[18] },
+    grade: { min: gradeRangeMetadata[0], max: gradeRangeMetadata[12] },
+  });
+  const [multiSelectInput, setMultiSelectInput] = useState({
+    Ethnicity: [],
+    Religion: [],
+    Gender: [],
+    Sexuality: [],
+  });
 
   const getCards = () => {
     base('Book').select({ view: 'Grid view' }).all()
       .then((records) => {
         setCards(records);
+        setFilteredCards(records);
       });
   };
 
-  useEffect(getCards, []);
+  useEffect(() => {
+    const validGradeTags = gradeRangeMetadata.slice(
+      gradeRangeMetadata.indexOf(rangeInput.grade.min),
+      gradeRangeMetadata.indexOf(rangeInput.grade.max) + 1,
+    );
+    const validAgeTags = ageRangeMetadata.slice(
+      ageRangeMetadata.indexOf(rangeInput.age.min),
+      ageRangeMetadata.indexOf(rangeInput.age.max) + 1,
+    );
+    console.log(rangeInput);
+    console.log(multiSelectInput);
+    console.log(cards);
+
+    setFilteredCards(cards.filter(
+      (record) => (record.fields.age_range.some((val) => validAgeTags.indexOf(val) !== -1)
+      && record.fields.grade_range.some((value) => validGradeTags.indexOf(value) !== -1))
+      && (multiSelectInput.Ethnicity.length === 0
+         || (record.fields['race/ethnicity'] !== undefined
+           ? record.fields['race/ethnicity'].some((value) => multiSelectInput.Ethnicity.indexOf(value) !== -1)
+           : true))
+      && (multiSelectInput.Religion.length === 0
+        || (record.fields.religion !== undefined
+          ? record.fields.religion.some((value) => multiSelectInput.Religion.indexOf(value) !== -1)
+          : true))
+      && (multiSelectInput.Sexuality.length === 0
+        || (record.fields.sexuality !== undefined
+          ? record.fields.sexuality.some((value) => multiSelectInput.Sexuality.indexOf(value)
+           !== -1)
+          : true
+        ))
+      && (multiSelectInput.Gender.length === 0
+        || (record.fields.gender !== undefined
+          ? record.fields.gender.some((value) => multiSelectInput.Gender.indexOf(value) !== -1)
+          : true)),
+
+    ));
+  }, [rangeInput, multiSelectInput]);
+
+  useEffect(() => { getCards(); }, []);
+  useEffect(() => {
+    console.log(filteredCards);
+  }, [filteredCards]);
 
   return (
     <div>
+      <RangeFilter
```

Comment:
As I specified on the rangeFilter component mage, the most external component listed out here would be better as a wrapper component for the two filters like "SearchFilterMenu" or something similar rather than declaring what looks like a single Range Filter. This will help with legibility!